### PR TITLE
<LibC> Add support for float in printf

### PR DESCRIPTION
**Pull request by akashkollipara**
- Added float support in printf. Without format specifier float prints 5 fractional places.
- Removed snprintf as it is outdated.

Issue: #154

### CODE REVIEW

**Comment by akashkollipara** [comment]
Functional Test passed on RISC-V

**Comment by akashkollipara** [comment]
AVR Target seems to miss out on float support... Pushing another path to fix AVR target.

**Comment by akashkollipara** [comment]
Tested on qemu AVR. Float engine works!

File: src/lib/libc/printf.c [hunk 1]
```diff
@@ -96,15 +95,41 @@ static int unumprint(const FILE *dev, bool en_stdout, unsigned long unum,
...
+}
+#endif
+
 int vprintf(const FILE *dev, bool en_stdout, const char *fmt, va_list args)
 {
 	int l_ret;
 	long num;
 	unsigned long unum;
-	char *str;
+	char *str, ch;
```
Comment (pranjalchanda08) [inline src/lib/libc/printf.c:125]: Use separate line for declaring the new variables.

**Review by pranjalchanda08 — CHANGES_REQUESTED** [review]
Minor changes else LGTM!

File: src/lib/libc/printf.c (hunk 1 above)
Comment (akashkollipara) [reply src/lib/libc/printf.c:125]: Any specific reason ?

Comment (pranjalchanda08) [reply src/lib/libc/printf.c:125]: It is as per the standards! 
image

Anyhow it is deactivated on the sonarcloud I feel.

**Review by pranjalchanda08 — APPROVED** [review]

**Pull request merged**